OCPBUGS#984 cluster identity should be granted 'Contributor' role to disk-encryption-set in azure install (#52550)
openshift/openshift-docs: modules/installation-azure-finalizing-encryption.adoc

[source,terminal]
----
$ az role assignment create --role "<privileged_role>" \// <1>

>     --assignee "<resourceGroup>" <2>

Reviewer: I think `resource_group_identity` is more easy reading.

----
<1> Specifies an Azure role that has read/write permissions to the disk encryption set. You can use the `Owner` role or a custom role with the necessary permissions.
<2> Specifies the identity of the cluster resource group, which is the `resourceGroup` value from the first step.
+

> . Grant a sufficient role assignment to the cluster principal so that it can access the encrypted storage. You may use the `Contributor` or `Owner` roles by running the following command:

Reviewer: Is there any difference between the new step and the above step 2? Per my understanding, they are duplicated.

Reviewer 2: What is being duplicated?

Reviewer: Read through https://github.com/openshift/openshift-docs/blob/0c7284cf61f086a4541ea159e4d1e2d6b9f170bc/modules/installation-azure-finalizing-encryption.adoc for step 2 and 3, actually they are doing the same things, right?

Reviewer 2: @jianlinliu , I see. It does appear to be duplicating a step. I've verified this locally. I've updated the bug to follow your instructions.

<1> Specifies an Azure role that has read/write permissions to the disk encryption set. You can use the `Owner` role or a custom role with the necessary permissions.
<2> Specifies the identity of the cluster resource group.
+
. Grant a role assignment to the cluster principal so that it can access the encrypted storage.
.. Obtain the principal identity by running the following command:
+
[source,terminal]
----

> $ PRINCIPAL_ID=$(az identity list --resource-group $CLUSTER_RESOURCE_GROUP '[0].principalId')

Reviewer: 1. The command never works.
```
$ az identity list --resource-group anli4102-t5phs-rg '[0].principalId'
unrecognized arguments: [0].principalId
```
2. Go through the whole page, the newly added section sounds doing the same thing as previous procedure 1 and 2. So I guess we can skip adding this new step, but re-use step 1 and 2.
3. But actually step 1 and 2 have wrong descriptions. In `user-managed encryption` case, there are two resource groups, one is pre-created resource group to save user-managed encryption set, another one is created by installer to save cluster resources. so the step should be like: `getting encryption set object id from user-managed encryption resource group` -> `get cluster identity principal id from cluster resource group` -> `assign cluster principal id permissions to access encryption set scope`

Author: @jhixson74 can you take a look at the syntax error here? What change do we need to make to the `az identity list` command?

Reviewer 2: There is no syntax error here as I can easily reproduce this on my machine. More likely a version issue. I will come up with another method to accomplish the same thing, though at some point it will come down to using the az tool or to avoid any sort of problem possibly logging into the azure portal.

Author: @jhixson74 @jianlinliu Thank you both, I've updated the PR to 1) avoid the syntax issues Jianlin is having and 2) move the new step to immediately after step 2 so that it makes more sense.

John, can you let me know if I need to rework or reword things to take into account Jianlin's point 3 above?

Reviewer 2: @bscott-rh , I have updated the JIRA bug with the new procedure.

Author: I saw that John - that's what I used in my latest commit.

<1> Specifies an Azure role that has read/write permissions to the disk encryption set. You can use the `Owner` role or a custom role with the necessary permissions.
<2> Specifies the identity of the cluster resource group.
+
. Obtain the `id` of the disk encryption set you created prior to installation by running the following command:
+
[source,terminal]
----

> $ az disk-encryption-set show -n <disk_encryption_set_name> <1>
>     --resource-group <resource_group_name> <2>

Reviewer 3: `\` is needed, extra ` `  on line below is style :) (space is to align under the first arg)
```suggestion
$ az disk-encryption-set show -n <disk_encryption_set_name> \ <1>
     --resource-group <resource_group_name> <2>
```

----
<1> Specifies the name of the disk encryption set.
<2> Specifies the resource group that contains the disk encryption set.

> The `id` will be in the format of `"/subscriptions/.../resourceGroups/.../providers/Microsoft.Compute/diskEncryptionSets/..."`.

Reviewer 3: ```suggestion
The `id` is in the format of `"/subscriptions/.../resourceGroups/.../providers/Microsoft.Compute/diskEncryptionSets/..."`.
```

+
. Obtain the identity of the cluster service principal by running the following command:
+
[source,terminal]
----

> $ az identity show -g <cluster_resource_group> <1>
>     -n <cluster_service_principal_name> <2> 
>     --query principalId --out tsv

Reviewer 3: ```suggestion
$ az identity show -g <cluster_resource_group> \ <1>
     -n <cluster_service_principal_name> \ <2> 
     --query principalId --out tsv
```

----
<1> Specifies the name of the cluster resource group created by the installation program.
<2> Specifies the name of the cluster service principal created by the installation program.

> The identity will be in the format of `12345678-1234-1234-1234-1234567890`.

Reviewer 3: ```suggestion
The identity is in the format of `12345678-1234-1234-1234-1234567890`.
```

> . Create a role assignment granting the cluster service principal `Contributor` privileges to the disk encryption set by running the following command:

Reviewer 3: ```suggestion
. Create a role assignment that grants the cluster service principal `Contributor` privileges to the disk encryption set by running the following command:
```

+
[source,terminal]
----

> $ az role assignment create --assignee <cluster_service_principal_id> <1>
>     --role 'Contributor' 
>     --scope <disk_encryption_set_id> <2>

Reviewer 3: ```suggestion
$ az role assignment create --assignee <cluster_service_principal_id> \ <1>
     --role 'Contributor' \
     --scope <disk_encryption_set_id> <2>
```
